src/truststore/_ssl_constants.py

```diff
@@ -4,8 +4,9 @@
 
 # Hold on to the original class so we can create it consistently
 # even if we inject our own SSLContext into the ssl module.
+SSLContextCopy = type("SSLContextCopy", (ssl.SSLContext,), dict(ssl.SSLContext.__dict__))
```

Comment:
Unfortunately this doesn't avoid the recursion problem, as most probably once the original `SSLContext` object gets replaced, we inevitably get into an infinite recursion when calling the super attributes.

That usually happens when multiple such injections occur.